### PR TITLE
Add misc enhancements

### PR DESCRIPTION
**Pull request by runlevel5**
A standard practice to verify the correctness of the ISO after downloading.

## Changes

* Verify checksum
* Add custom Alpine 3.12 QEMU container image
* Minor refactoring on shell scripts

### CODE REVIEW

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -2,7 +2,19 @@
 set -e
 cd /build/
 
-if [ ! -e "alpine-virt-3.8.0-x86_64.iso" ]; then
-    wget http://dl-cdn.alpinelinux.org/alpine/v3.8/releases/x86_64/alpine-virt-3.8.0-x86_64.iso
+MAJOR_VERSION="${MAJOR_VERSION:-3.8}"
+MINOR_VERSION="${MINOR_VERSION:-0}"
```
Comment (runlevel5) [inline docker-entrypoint.sh:6]: I take the opportunity to refactor the versioning code. This opens up new way to build different version of Alpine by passing in custom major/minor version Unix environment variables.

File: docker-entrypoint.sh [hunk 2]
```diff
@@ -2,7 +2,19 @@
...
+CHECKSUM_FILENAME="$ISO_FILENAME.sha512"
+URL="http://dl-cdn.alpinelinux.org/alpine/v$MAJOR_VERSION/releases/x86_64"
+
+if [ ! -e $ISO_FILENAME ]; then
+    wget $URL/$ISO_FILENAME
+fi
+
+if [ ! -e "${CHECKSUM_FILENAME}" ]; then
+    wget $URL/$CHECKSUM_FILENAME
+    sha512sum --strict -c $CHECKSUM_FILENAME
```
Comment (runlevel5) [inline docker-entrypoint.sh:17]: `--strict` means exit with non-zero status if checksum does not match

**Comment by runlevel5** [comment]
I also unrelated question why Alpine 3.8 was chosen instead of 3.12? cc @TalonLaurens

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,31 @@
+FROM alpine:3.12
```
Comment (runlevel5) [inline Dockerfile:1]: The `justincormack/alpine-qemu` image is alpine 3.3 which does not ship with new version of sha512 that supports the `--strict` flag. 

IMHO it'd be best to ask the users to build this up-to-date container image

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  qemu-system-mips64 \
+  qemu-system-moxie \
+  qemu-system-s390x \
+  qemu-system-tricore \
+  qemu-system-mips \
+  qemu-system-microblazeel \
+  qemu-system-sh4 \
+  qemu-system-sparc \
+  expect \
+  cdrkit
```
Comment (runlevel5) [inline Dockerfile:31]: This package offers `mkisofs` tooling

File: Makefile [hunk 5]
```diff
@@ -1,2 +1,3 @@
 build:
-	docker run --rm --volume $(shell pwd):/build justincormack/alpine-qemu /build/docker-entrypoint.sh
\ No newline at end of file
+	docker build . -t qemu-alpine
+	docker run --rm --volume $(shell pwd):/build:Z qemu-alpine /build/docker-entrypoint.sh
```
Comment (runlevel5) [inline Makefile:3]: The `Z` flag is added to grant full access to current folder on Linux system with SELinux. If not, users would get file access denied error.

File: .dockerignore [hunk 6]
```diff
@@ -0,0 +1,3 @@
+disk.raw
+disk.raw.tar.gz
+*.iso
```
Comment (runlevel5) [inline .dockerignore:3]: This would ensure that those huge files won't be streamed to docker as part of the building step.

**Comment by TalonLaurens** [comment]
Thank you for this PR! Most of your changes look very reasonable. However the main maintainer of this repo is currently on parental leave until mid August, so we can't really verifyit fully until the.

As to why we used Alpine 3.8 in stead of 3.12, I think it was simply the latest stable version when we started working on this project, no specific reason.

**Comment by Eun** [comment]
Thank you for your PR, I am kinda surprised that somebody is using this :sweat_smile:

**Pull request merged**